src/shared/lib/AnnotationUtils.ts

```diff
@@ -45,18 +50,131 @@ export function geneToMyCancerGenome(myCancerGenomes:IMyCancerGenome[]):IMyCance
 }
 
-export function geneAndProteinPosToHotspots(hotspots:HotspotMutation[]):{[key:string]: boolean}
+export function indexHotspots(hotspots:HotspotMutation[], defaultHotspotType: string = "hotspot"): IHotspotIndex
 {
-    // key: geneSymbol_proteinPosition
-    // (protienPosition: start[_end])
-    const map: {[key:string]: boolean} = {};
+    const index:IHotspotIndex = {};
 
-    // create a map for a faster lookup
-    hotspots.forEach(function(hotspot:HotspotMutation) {
-        const positions = hotspot.residue.match(/[0-9]+/g) || []; // start (and optionally end) positions
-        const key = [hotspot.hugoSymbol.toUpperCase()].concat(positions).join("_");
-        map[key] = true;
+    hotspots.forEach((hotspot:HotspotMutation) => {
+        const geneSymbol = hotspot.hugoSymbol.toUpperCase();
+        let geneIndex = index[geneSymbol];
+
+        if (!geneIndex) {
+            geneIndex = index[geneSymbol] = {};
+        }
+
+        const hotspotType = hotspot.type || defaultHotspotType;
+        let lookup = geneIndex[hotspotType];
+
+        if (!lookup)
+        {
+            lookup = {
+                hotspotMutations: [],
+                hotspotSet: new HotspotSet([]) // this is a placeholder to avoid TS errors.
+                                               // overwritten by the initHotspotSets function.
+            };
+
+            geneIndex[hotspotType] = lookup;
+        }
+
+        lookup.hotspotMutations.push(hotspot);
     });
 
-    return map;
+    initHotspotSets(index);
+
+    return index;
+}
+
+export function initHotspotSets(hotspotIndex: IHotspotIndex)
+{
+    _.keys(hotspotIndex).forEach((geneSymbol:string) => {
+        const geneIndex = hotspotIndex[geneSymbol];
+
+        _.keys(geneIndex).forEach((hotspotType:string) => {
+            const lookup = geneIndex[hotspotType];
+
+            const intervals = lookup.hotspotMutations.map(
+                (hotspot:HotspotMutation):[number, number] => {
+                    // start (and optionally end) positions
+                    const positions = hotspot.residue.match(/[0-9]+/g) || [];
+
+                    let start = -1;
+                    let end = -1;
+
+                    if (positions.length > 0) {
+                        start = parseInt(positions[0]);
+                        end = start;
+                    }
+
+                    // overwrite end if exists
+                    if (positions.length > 1) {
+                        end = parseInt(positions[1]);
+                    }
+
+                    return [start, end];
+                }
+            );
+
+            lookup.hotspotSet = new HotspotSet(intervals);
+        });
+    });
+}
+
+export function isHotspot(mutation:Mutation, index:IHotspotIndex):boolean
+{
+    let isHotspot = false;
+    let hotspotType:string|null = null;
+    const mutationType = mutation.mutationType && mutation.mutationType.toLowerCase();
+
+    if (mutationType.indexOf("missense") > -1) {
```

Comment:
Good catch! Actually the problem here is the `Mutation` model definition in the generated API client. The model assumes that those fields always exist, but in practice I don't think that's the case.